### PR TITLE
Minor imprvements considering the logs relevant to the subscription rate-limiting

### PR DESCRIPTION
**Pull request by slahirucd7**
$subject

Relates with:
https://github.com/wso2-enterprise/choreo/issues/27601
https://github.com/wso2-enterprise/choreo/issues/26811

### Issues

Fixes #

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java [hunk 1]
```diff
@@ -348,6 +348,9 @@ public AuthenticationContext authenticate(RequestContext requestContext) throws
                             requestContext.addMetadataToMap("ratelimit:organization",
                                     APIConstants.SUPER_TENANT_DOMAIN_NAME);
                         }
+                        log.debug("Organization ID: " + requestContext.getMetadataMap().get("ratelimit:organization")
+                                + ", SubscriptionId: " + subscriptionId + ", SubscriptionPolicy: " + subPolicyName
+                                + " will be evaluated for subscription rate-limiting");
```
Comment (renuka-fernando) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java:353]: shall we check if isDebug is enabled before generating this long string?

**Review by renuka-fernando — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[succeeded]  : dev-deployment-v2 : 20240509.16

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : dev-deployment-v2 : 20240509.16

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : dev-deployment-v2 : 20240509.16

**Comment by choreo-cicd** [comment]
[failed] Dataplane(EastUS) cluster : stage-deployment-v2 : 20240513.3

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : stage-deployment-v2 : 20240513.3

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20240513.4

**Comment by choreo-cicd** [comment]
[failed] Dataplane(CDP2-WU2) cluster : stage-deployment-v2 : 20240513.4

**Comment by choreo-cicd** [comment]
[failed] Dataplane(EastUS) cluster : stage-deployment-v2 : 20240513.4

**Comment by choreo-cicd** [comment]
[failed] Dataplane(CDP2-WU2) cluster : stage-deployment-v2 : 20240513.5

**Comment by choreo-cicd** [comment]
[failed] Dataplane(EastUS) cluster : stage-deployment-v2 : 20240513.5

**Comment by choreo-cicd** [comment]
[] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20240513.6

**Comment by choreo-cicd** [comment]
[] Controlplane cluster : stage-deployment-v2 : 20240513.6

**Comment by choreo-cicd** [comment]
[] Dataplane(CDP2-WU2) cluster : prod-deployment-v2 : 20240515.2

**Comment by choreo-cicd** [comment]
[] Dataplane(CDP2-WU2) cluster : prod-deployment-v2 : 20240515.2

**Comment by choreo-cicd** [comment]
[] Dataplane(EastUS) cluster : prod-deployment-v2 : 20240515.2

**Comment by choreo-cicd** [comment]
[] Dataplane(EastUS) cluster : prod-deployment-v2 : 20240515.2

**Comment by choreo-cicd** [comment]
[] Controlplane cluster : prod-deployment-v2 : 20240515.2

**Comment by choreo-cicd** [comment]
[] Controlplane cluster : prod-deployment-v2 : 20240515.2

**Comment by choreo-cicd** [comment]
[] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20240515.2

**Comment by choreo-cicd** [comment]
[] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20240515.2

**Pull request merged**